### PR TITLE
Add "failFastFailureRegexes" property, like anti-"ignoreFailureRegexes"

### PR DESCRIPTION
**Pull request by idyatlov**
Added `failFastFailureRegexes` configuration property

- Tests that have failed with stack traces that match that property wouldn't be rerun
- It applies to both failed and uncompleted tests
- This has higher priority than uncompletedRetriesQuota or amount of strict runs

### CODE REVIEW

File: cli/src/main/kotlin/com/malinskiy/marathon/cli/args/FileConfiguration.kt [hunk 1]
```diff
@@ -38,6 +38,7 @@ data class FileConfiguration(
     var includeSerialRegexes: Collection<Regex>?,
     var excludeSerialRegexes: Collection<Regex>?,
     var ignoreFailureRegexes: Collection<Regex>?,
+    var strictFailureRegexes: Collection<Regex>?,
```
Comment (jeksor) [inline cli/src/main/kotlin/com/malinskiy/marathon/cli/args/FileConfiguration.kt]: I know all the other fields don't have any comments, but I think we should start adding them.
The number of various parameters is getting bigger and bigger.
It's worth mentioning `This has higher priority than uncompletedRetriesQuota or amount of strict runs`.

Anothing thing, the current naming uses the word `strict`. The closest analogy is strict run, which means "run until it's green or failed X times". Although the current field has the opposite logic "do not rerun if tests failure contains one of the issues, an event it's under the strict run".

Current me if I am wrong, please.

Therefore I would suggest eliminating a potential misunderstanding.
Maybe `failfastFailureRegexes`, `rerunlessFailureRegexes`or something that reflects the logic behind it.

File: buildSrc/src/main/kotlin/Versions.kt [hunk 2]
```diff
@@ -5,7 +5,7 @@ object Versions {
     val ktor = "1.6.5"
 
     val ddmlib = "30.1.2"
-    val dexTestParser = "830520963019a6cefada34fc5eb396003c1468d5" // contains patch https://github.com/linkedin/dex-test-parser/pull/46
+    val dexTestParser = "2.3.4"
```
Comment (jeksor) [inline buildSrc/src/main/kotlin/Versions.kt:8]: I hope the new version doesn't have any surprises :hiddenpain:

Comment (idyatlov) [reply buildSrc/src/main/kotlin/Versions.kt:8]: It actually has not so many commits, our fix is included in [2.2.1](https://github.com/linkedin/dex-test-parser/releases) and now it's 2.3.4, but they have fixed a quite important [bug](https://github.com/linkedin/dex-test-parser/issues/72) which could affect us.

File: cli/src/main/kotlin/com/malinskiy/marathon/cli/args/FileConfiguration.kt (hunk 1 above)
Comment (idyatlov) [reply cli/src/main/kotlin/com/malinskiy/marathon/cli/args/FileConfiguration.kt]: failfastFailureRegexes looks good, will change

**Review by jeksor — APPROVED** [review]

**Pull request merged**